### DISCUSSION TITLE
Please make it work for Silicon!

### DISCUSSION THREAD

**Discussion by pireba** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#70] [Ideas]
I was looking for ages for an app like this, now I finally found it but am in possession of an M1 MacBook 😀
The icon and name are great by the way, very stylish. Greetings from Bremen, Germany ✌️

**Comment by thisbansal** [comment]
I'm pretty sure that battery health management that comes with macOS Big Sur already does this to keep the battery in healthy state for as long as possible!

**Reply by pireba** [reply]
I know about this thing but I don't really get what it's doing. There ist no way to control it by myself. I would like to have full control over the charging. The best thing would be a menubar app, where I can select "charge" and "don't charge" and maybe set an automatic that manages, that the battery charges until let's say 80% and then uses the battery until let's say 30% and then it automatically starts charging until 80% and so on... Whatever, let's hope that this health management does it good 😄

**Reply by Cheule** [reply]
The battery mgmt. that comes with Big Sur will allow the battery to charge to 100% if the battery is plugged in for days to weeks.  

What I've seen, and someone correct me if I'm mistaken, is that it works similarly to the implementation for iOS.  The system learns when you are likely to disconnect from power (i.e. when you leave in the morning).  Then the system allows the battery to charge to 80% and sits at 80% until about 3 hours before you usually disconnect from power.  Then it charges to 100% to be ready.

The issue lies with people like myself that *never* disconnect from power routinely.  And this is where Al Dente is especially useful.

**Reply by pireba** [reply]
Thats how I understand that as well and your problem is exactly my problem. Thats why I would love to have this app working on M1 :)

**Comment by MatthiasKerbl** [comment]
The time has come: AlDente for Apple Silicon (M1 MacBooks) is out now. We thank everyone who has helped us here in the forum and we hope you like it. That's not all, we have also started working on a pro version of AlDente with advanced features several weeks ago. Since we have put a lot of hours into developing the app, it will cost a little. However, there will always be this free and open-source version of AlDente updated with the new M1 SMC keys.  If you have any feature requests or ideas for AlDente Pro, let us know. If you want to get notified, when AlDente Pro comes out, just enter your email at https://apphousekitchen.com/aldente/ In the meantime, stay safe and keep your battery fresh. David & Matthias

**Comment by gubnota** [comment]
Just Installed the latest update for M1 Macs and it seems there is a bug: when you drag to pointer to the desired value (80% e.g.) it puts the value down to 3 or 0% (somehow), and even when you manage to set the value correct when you relaunch the app it still shows 0 or 3%. When you launch the app from the command line it shows how it tries to set the value the the terminal output. When you try to uninstall it completely (removing the launcher from /Library and all plists) the wrong value is still there (I am not sure where😂 but you got the gist).

Moreover, when you reboot the computer, and discharge the battery, let say, to 98% then you connect the charger at it shows 100% instead of 98% (maybe this glitch or bug is not really related to AlDente).

It would be nice to have command line interface to interact with the tool, not just this UI which seems to be Intel arch and a little bit buggy.

**Comment by davidwernhart** [comment]
Hey @gubnota 
I am sorry to hear about your experience.
Sadly, I have not managed to replicate the bugs yet.
Can you maybe specify a few more details?
Are you on an M1 or Intel Machine? Do you set your desired charge value using the text box or the slider? You mean the wrong value is still there after you uninstall and reinstall AlDente?

> Moreover, when you reboot the computer, and discharge the battery, let say, to 98% then you connect the charger at it shows 100% instead of 98% (maybe this glitch or bug is not really related to AlDente).

This is most likely a MacOS feature, as the OS constantly adjusts the displayed max percentage based on battery stats. This is also the reason why, with the old Intel key, the set charge value was wrong by about 3% most of the time.

Also, if you are on an M1 mac, AlDente is compiled natively for Apple silicon. You can check so with activity monitor.

Best regards, David

**Reply by gvalencia4** [reply]
I also had the issue where the value would reset to 3 after a second or so, no matter if I used the slider or the text box. I am using a M1 Mac. The issue initially went away after I uninstalled, restarted, and installed again. After I closed the app and opened it again, the 3% issue came back.

**Reply by gubnota** [reply]
Thank you for your swift reply, @davidwernhart, and @gvalencia4 for better explanation. Same M1 here, I think it would be easier to track the bug while using some logging or command line prompt. When you start it from command line you can see it is trying to change some values, apparently it doesn't work. It might be something related to system Integrity protection. Because, XtraFinder plugin which I bought also works with Intel-based Macs on the latest version of 11.2 so far, whilst doesn't work even with SIP turned off (I tried and eventually compiled my own Finder extention instead🥸). It seems like GUI should interact and call back on its value to show a message something like "Oops, I can't change that yet. Give up for a week then try an updated version".

**Comment by pireba** [comment]
First if all: big big thanks for updating this for M1, that's great work and as pathetic as it sounds - it's something simple the world really needs since it will reduce the replacing of batteries. I just want to say that for me it works perfectly. I am running macOS 11.2 on a MacBook Pro M1 and the battery is constantly at exactly 70% as it's told to for two days now.

**Reply by mikebore** [reply]
That's good to hear. 
As I understand it this is a mod to an SMC setting.
What happens when the SMC is reset?
On M1 Macs SMC is reset simply by shutting down for 30 secs and restarting, which could happen quite often.
Thanks

**Reply by davidwernhart** [reply]
A smc reset should do no harm, it basically resets all values to it's default (which for AlDente's key is: no charge inhibit, no active discharge). If you start AlDente again after a reset, these keys get modified again for the tool to to it's job.

**Reply by mikebore** [reply]
Thanks for quick reply!
I'll give it a go.

**Reply by Jens-Oldenburg** [reply]
Great work, the last version 2.1 now works as expected on my M1. Thanks a lot for your support.
